modules/maps.js

```diff
@@ -86,6 +77,217 @@ const uniqueMaps = {
     }
 };
 
+class MappingProfile {
+    priorities;
+    mods = [];
+    minLevel;
+    baseLevel;
+    optimalLevel;
+    preferredBiome;
+    name;
+
+    static priorities = Object.freeze({
+        size: 'size',
+        loot: 'loot',
+        diff: 'difficulty',
+        mod: 'mod',
+        minLevel: 'minLevel',
+        baseLevel: 'baseLevel',
+        optimalLevel: 'optimalLevel',
+        biome: 'biome'
+    });
+    static profiles = Object.freeze({
+        farming: {
+            // * farming: {(FA [desperation?] > LMC > map size) > level > loot > extraLevel} > diff*
+            required: [
+                MappingProfile.priorities.mod
+            ],
+            optional: [
+                MappingProfile.priorities.biome,
+                MappingProfile.priorities.size,
+                MappingProfile.priorities.baseLevel,
+                MappingProfile.priorities.loot,
+                MappingProfile.priorities.optimalLevel,
+                MappingProfile.priorities.diff
+            ],
+            mods: ["cache", "fa"],
+        },
+        prestige: {
+            // * prestige: (zoneLevel > FA) > map size > P) [not sure which] > diff* > loot > 10 extraLevels [JK, but who knows?]
+            required: [
+                MappingProfile.priorities.minLevel
+            ],
+            optional: [
+                MappingProfile.priorities.mod,
+                MappingProfile.priorities.size,
+                MappingProfile.priorities.diff,
+                MappingProfile.priorities.biome,
+                MappingProfile.priorities.loot,
+                MappingProfile.priorities.optimalLevel
+            ],
+            mods: ['p', 'fa']
```

Comment:
This case is handled by the logic that chooses any mod if we can't afford the map with the required mod.
Anyway, i'll improve this logic while adding the "limit fragment spending" option.

Reply:
Hm... Could you please elaborate?

I mean, from what I understood, it'll correctly start with "minLvl", then, with a bit more frags, "minLvl + FA" (correct).
But if it has more frags than that, it'll go for "minLvl + P" instead of "minLvl + FA + size".

Reply:
You are correct, currently FA -> size -> P is not supported

Reply:
Fixed